extensions/axon/runtime/src/main/java/io/quarkus/axon/runtime/AxonArcBeanResolverFactory.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.axon.runtime;
+
+import java.lang.reflect.Executable;
+import java.lang.reflect.Parameter;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+
+import io.quarkus.arc.runtime.BeanContainer;
+
+@ApplicationScoped
+public class AxonArcBeanResolverFactory implements ParameterResolverFactory {
+
+    private static BeanContainer beanContainer;
+
+    public AxonArcBeanResolverFactory() {
+    }
+
+    // This class will be loaded by the Java service loader and there is no dependency injection possible (?)
+    // so the beanContainer should be set in a static way.
+    static void setBeanContainer(BeanContainer beanContainer) {
```

Comment:
Can you please explain what you are trying to do here that requires the service loader?

Wouldn't something like https://github.com/quarkusio/quarkus/blob/7e9330df7f9b7f66887d94d15cd073379fff253a/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/VertxWebRecorder.java#L46 work?